dremio_client/model/endpoints.py

```diff
@@ -284,6 +326,70 @@ def user(token, base_url, uid=None, name=None, ssl_verify=True):
         return _get(endurl, token, endurl, ssl_verify)
 
 
+def create_role(token,base_url,json ,ssl_verify=True):
+    return _post(base_url + "/api/v3/role", token, json, ssl_verify=ssl_verify)
+
+
+def get_role(token, base_url, uid, name, ssl_verify=True):
+    if uid is None and name is None:
+        raise TypeError("both id and name can't be None for a user call")
+    if uid is not None :
+        return _get(base_url + "/api/v3/role/{}".format(uid), token, ssl_verify=ssl_verify)
+    else:
+        return _get(base_url + "/api/v3/role/by-name/{}".format(name), token, ssl_verify=ssl_verify)
+
+
+def delete_role(token, base_url, uid , ssl_verify=True):
+    return _delete(base_url + "/api/v3/role/{}".format(uid) , token, ssl_verify=ssl_verify)
+
+
+def update_role(token, base_url, uid, json, ssl_verify=True):
+    return _put(base_url + "/api/v3/role/{}".format(uid), token, json, ssl_verify=ssl_verify)
+
+def get_all_roles(token, base_url ,startIndex=None , count=None ,query=None  , ssl_verify=True):
+    if startIndex is not None and count is not None and query is not None:
```

Comment:
this (and the ones below it) can be simplified.
May I suggest somethign like (i didn't test this):
```
params = dict()
if startIndex is not None:
  params['startIndex'] = startIndex
...repeat for other params
query = '&'.join("{}={}".format(k,v) for k,v in params)
return _get(base_url + "/api/v3/role{}".format(("?" + query) if query else "" )
```